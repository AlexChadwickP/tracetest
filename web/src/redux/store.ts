import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import {testAPI} from '../services/TestService';

export const store = configureStore({
  reducer: {
    [testAPI.reducerPath]: testAPI.reducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
