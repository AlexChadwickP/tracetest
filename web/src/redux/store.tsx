import {Action, configureStore, ThunkAction} from '@reduxjs/toolkit';
import TestAPI from 'redux/apis/TraceTest.api';

export const store = configureStore({
  reducer: {
    [TestAPI.reducerPath]: TestAPI.reducer,
  },
  middleware: getDefaultMiddleware => getDefaultMiddleware().concat(TestAPI.middleware),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
