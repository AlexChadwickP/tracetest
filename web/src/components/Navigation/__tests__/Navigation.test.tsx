import {render} from '@testing-library/react';
import {ReduxWrapperProvider} from '../../../redux/ReduxWrapperProvider';
import AnalyticsProvider from '../../Analytics';
import Router from '../Router';

test('Router', async () => {
  const {container} = render(
    <AnalyticsProvider>
      <ReduxWrapperProvider>
        <Router />
      </ReduxWrapperProvider>
    </AnalyticsProvider>
  );

  expect(container).toMatchSnapshot();
});
