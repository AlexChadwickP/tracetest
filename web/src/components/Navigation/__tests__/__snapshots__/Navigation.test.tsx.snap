// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router 1`] = `
<div>
  <div
    data-cy="not_initialized"
  />
</div>
`;
